tests/wasm_tests/wasm_tests.cpp

```diff
@@ -830,5 +830,96 @@ BOOST_FIXTURE_TEST_CASE( check_table_maximum, tester ) try {
 } FC_LOG_AND_RETHROW()
 #endif
 
+BOOST_FIXTURE_TEST_CASE( test_db, tester ) try {
+   produce_blocks(2);
+   
+   create_accounts( {N(tester)} );
+   produce_block();
+   
+   set_code(N(tester), test_api_wast);
+   //   set_code(N(tester), test_api_abi);
+   
+   produce_blocks(1);
+   
+   {
+      signed_transaction trx;
+      trx.actions.emplace_back(vector<permission_level>{{N(tester), config::active_name}},
+                               test_api_action<TEST_METHOD("test_db", "primary_i64_general")> {});
+      
+      set_tapos(trx);
+      trx.sign(get_private_key(N(tester), "active"), chain_id_type());
+      push_transaction(trx);
+      produce_block();
+      
+      BOOST_REQUIRE_EQUAL(true, chain_has_transaction(trx.id()));
+   }
+
+   {
```

Comment:
Minor issue - It seems like all the block down from this line are very similar. The only thing that different is function name that gets called. So, maybe it's possible to just do a loop over function names.
Also there is a function tester::push_action (libraries/testing/tester.cpp, usage example - tests/wasm_tests/identity_tests.cpp) which might be used here to reduce boilerplate code.